src/mon/OSDMonitor.cc

```diff
@@ -8321,6 +8321,9 @@ bool OSDMonitor::prepare_command_impl(MonOpRequestRef op,
     wait_for_finished_proposal(op, new Monitor::C_Command(mon, op, 0, rs,
 					      get_last_committed() + 1));
     return true;
+  } else if (prefix == "osd get-require-min-compat-client") {
```

Comment:
This is a read-only command. It belongs in `OSDMonitor::preprocess_command()` instead.

Reply:
Well the responsibilities of the code in that file are a bit convoluted. (Functions with thousands of lines?!) so sorry, for not seeing that. But updated per your instructions.